### PR TITLE
[Pending] New Resource: aws_workspaces_workspace

### PR DESCRIPTION
**Pull request by atsushi-ishibashi**
#434 
I have to crawl a website which only suits IE so I'd like to add workspaces resource.
AWS API provides `CreateWorkspaces` which can create some workspaces at a time, but it's complicated to write multi workspaces in one resource like below.
```
resource "aws_workspaces_workspaces" "hoge" {
  workspace = {
    a = a1
    b = b1
    ....
  }

  workspace = {
    a = a2
    b = b2
    ....
  }
}
```
So I'm going to add `aws_workspaces_workspace` and for each `CreateWorkspaces` works with one element.

### CODE REVIEW

**Comment by atsushi-ishibashi** [comment]
I got the below error and I couldn't api to register the directory to workspaces.
```
The specified directory could not be found in the specified region.
```
I asked aws support but they said that they could't support api to register so you had to do it from web console.

So this PR is pending until aws supports it.

**Comment by liamjbennett** [comment]
Reached the same issue here: https://github.com/hashicorp/terraform/pull/7806

**Comment by bcornils** [comment]
looks like we have a check failing and the branch has conflicts. Has anyone had time to assess or review criticality of the failure?

**Comment by sstarcher** [comment]
@atsushi-ishibashi are you going to be able to pick this back up or do you need someone to take it over.

**Comment by bflad** [comment]
Closing this PR due to lack of response. I should note that the `aws_workspaces_bundle` data source is separately being merged in #3243 and that will be released in version 1.40.0 of the AWS provider, likely middle of this week. 👍

**Comment by cmaxwellau** [comment]
@atsushi-ishibashi @liamjbennett

I work for AWS, and have a customer that would like to use terraform for workspaces. 

Would one of you be able to concisely describe the issue for me so that I can pass it on to the service teams?
xCam

**Comment by tamsky** [comment]
@cmaxwellau
> Would one of you be able to concisely describe the issue for me so that I can pass it on to the service teams?

tl;dr:
The AWS service team must create a new API endpoint, which currently doesn't exist (endpoint name is up to the service team, but this is my guess as to what it would be called:) `CreateWorkspaceDirectories`.

----

Given that there hasn't been a reply from anyone else in the last 90 days,
I'll try my best to give you my (admittedly non-expert) understanding of the issue.

To anyone reading this, it's highly likely that you know this issue better than I.
Please correct me if you see anything wrong here:

AIUI, in order to create a workspace, using the API call `CreateWorkspaces`[1] one must first have _already_ registered a workspace directory and provide a valid `DirectoryId` in the API call payload.

The act of "registering a workspace directory" is currently described in the adminguide at [2].
This appears to be an interactive process that demands a human use a web browser to click and type on the AWS web console.

Terraform automated tests require api-based environment creation/destruction, so we can't automate tests unless we have an API endpoint for creating a Workspace Directory.

- [1] https://docs.aws.amazon.com/workspaces/latest/api/API_CreateWorkspaces.html
- [2] https://docs.aws.amazon.com/workspaces/latest/adminguide/register-deregister-directory.html

**Comment by tamsky** [comment]
I see now that @liamjbennett replied over in
https://github.com/hashicorp/terraform/issues/10794#issuecomment-435563758

There was a post from an AWS rep on the forum thread [1]:
> Our internal team has confirmed that there is already an existing feature request placed on our system to address this requirement.
> At this moment, we don't have an ETA on when this feature will be released. 

- [1] https://forums.aws.amazon.com/thread.jspa?messageID=749994#749994
